lib/rules/no-throw-literal.js

```diff
@@ -32,12 +32,13 @@ module.exports = {
             ThrowStatement(node) {
                 if (!astUtils.couldBeError(node.argument)) {
                     context.report({ node, message: "Expected an object to be thrown." });
-                } else if (node.argument.type === "Identifier") {
-                    if (node.argument.name === "undefined") {
+                } else if (node.argument.type === "Identifier" && node.argument.name === "undefined") {
+                    const variable = astUtils.getVariableByName(context.getScope(), "undefined");
+
+                    if (!variable.references.some(ref => ref.isWrite())) {
```

Comment:
We cannot reassign to `undefined` variable since ES5. The runtime just ignores reassignment of `undefined` global variable.

Reply:
Oh well, do we need to close this PR?

Reply:
If it's about redeclaration/reassignment, yes. Though, we can shadow it.